### PR TITLE
Development

### PR DESCRIPTION
**Pull request by dipto0321**
Hi,
We have completed this project. The game is currently just 1 player (human vs computer) but it has two levels (easy and hard).. In the hard level, no human being, ever, will be able to defeat the computer

### CODE REVIEW

File: assets/js/app.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+];
+
+// Collect all cells
+let cells = [...document.getElementsByClassName("cell")];
+
+let game = new Game();
+let human = game.human;
+let computer = game.computer;
+
+for (let i = 0; i < cells.length; i++) {
```
Comment (heyalvaro) [inline assets/js/app.js]: [ESLint also dislikes the unary operators](https://eslint.org/docs/rules/no-plusplus#disallow-the-unary-operators--and----no-plusplus)

Comment (dipto0321) [reply assets/js/app.js]: Hi @heyapricot , 
I think unary is okay for `forLoop` iteration. 🙂 
[Eslint ](https://prnt.sc/mmz9vm) also agree with me.

**Comment by rvvergara** [comment]
# It works! slightly_smiling_face
 # But, it doesn't satisfy the requirements upside_down_face
 
 From the Odin Project's requirements:
 
     1. You’re going to store the gameboard as an array inside of a **Gameboard object**, so start there! Your players are also going to be stored in objects… and you’re probably going to want an **object to control the flow of the game itself**.
 
 
     * There is no Gameboard Object - Answer: The Game class is our GameBoard object. Its constructor initializes also the board (which is an array). 
 
     * The Game object doesn't control the flow of the game -> We now included modules for domControls and gameControls. But we kept the classes for Game and Player. 
 
 
 2.1. Your main goal here is to have as **little global code as possible**. Try tucking everything away inside of a module or factory. **Refer below**
 
     * I'm ok with the use of the Class notation, but there are no modules -> **We maintained Game and Player classes (and subclass Computer for Player). But for game-related controls and dom-related controls we put them all into their own modules in helper.js** 
 
     * The files depend **a lot** on global code. Particularly the helpers. -> **we fixed this already, only 2 global variables now, game and name**
  
     1. Set up your HTML and write a JavaScript function that will render the contents of the gameboard array to the webpage (for now you can just manually fill in the array with "X"s and "O"s)
     * This is missing -> **Missing before but not anymore** :D
 
     1. Clean up the interface to allow players to put in their names, include a button to start/restart the game and add a display element that congratulates the winning player!
     * This is missing -> **Not missing anymore** :D 
 
 
 Please fix and when finished send a new code review request. Thanks

**Comment by rvvergara** [comment]
> # Why are you reluctant to use modulesquestion -> We are not reluctant to use anything. If you mean "modules" as in using webpack to separate the files into "modules" then I guess this project doesn't yet fall within that category (it's coming in the lesson after this project). 
> 
>     1. I see that you have a helpers file but why not just define a Gameboard Object. Remember that functions and modules (as defined in The Odin project) are also objects. Why not create a Gameboard Object/Class that contains the methods for writing to the internal gameboard array?. Same with Game. The project is hard to understand/debug because I have to jump between files.
- The project is pretty clear -> The Game class is where you start because it's the mother of the app.. The game class upon instantiation creates the players (Human and Computer) as you see in its constructor. Then we have the game cycle method in the Game class which calls the gameControls object (which is in helpers.js) for the ff methods:
-gameControls.playerTurn -> that controls the action of the human and computer players.
-gameControls.endGame -> that checks whether someone has won or it's a draw (by utilizing the checkWin and checkDraw functions) and does the appropriate action.
> 
>     2. I also want you to separate DOM manipulation from the game logic
In this case please let us know which lines we should still separate. We have gone through this app line by line to separate all that should be and can be separated. So those that are left are left there because it's the best place for them (certain pieces of code) to be. The separation is pretty clear in helpers.js, we separated the domControls from gameControls. 
> 
> 
> Before sending a new code review request please contact me on Crisp to clarify the improvements. Thank you

**Pull request merged**